zipline/assets/assets.py

```diff
@@ -338,6 +446,8 @@ def spawn_asset(self, identifier, **kwargs):
         self.identifier_cache[identifier] = asset
         if asset.symbol is not '':
             self.sym_cache.setdefault(asset.symbol, []).append(asset)
+        if isinstance(asset, Future) and asset.root_symbol is not '':
+            self.future_cache.setdefault(asset.root_symbol, []).append(asset)
```

Comment:
There's not really any problem with the instance check. I was just thinking that, since we already have a flow where we know the Asset is a Future, we can just put this insertion in that flow. We would still insert it in to .cache with all other assets. The current state of the logic is fine.